orange/src/main/java/com/hartwig/hmftools/orange/algo/purple/TumorStatsFactory.java

```diff
@@ -0,0 +1,89 @@
+package com.hartwig.hmftools.orange.algo.purple;
+
+import java.util.List;
+
+import com.hartwig.hmftools.common.purple.GermlineStatus;
+import com.hartwig.hmftools.common.sv.EnrichedStructuralVariant;
+import com.hartwig.hmftools.common.sv.EnrichedStructuralVariantFactory;
+import com.hartwig.hmftools.common.sv.StructuralVariantType;
+import com.hartwig.hmftools.common.variant.VariantTier;
+import com.hartwig.hmftools.common.variant.VariantType;
+import com.hartwig.hmftools.datamodel.purple.ImmutableTumorStats;
+import com.hartwig.hmftools.datamodel.purple.TumorStats;
+
+import org.jetbrains.annotations.NotNull;
+
+public class TumorStatsFactory
+{
+
+    @NotNull
+    public static TumorStats compute(@NotNull PurpleData purpleData, boolean convertGermlineToSomatic)
+    {
+        return ImmutableTumorStats.builder()
+                .maxDiploidProportion(purpleData.purityContext().score().maxDiploidProportion())
+                .hotspotMutationCount(hotspotMutationCount(purpleData, convertGermlineToSomatic))
+                .hotspotStructuralVariantCount(hotspotStructuralVariants(purpleData))
+                .smallVariantAlleleReadCount(smallVariantAlleleReadCount(purpleData))
+                .structuralVariantTumorFragmentCount(structuralVariantTumorFragmentCount(purpleData))
+                .bafCount(bafCount(purpleData))
+                .build();
+    }
+
+    private static int structuralVariantTumorFragmentCount(@NotNull PurpleData purpleData)
+    {
+        List<EnrichedStructuralVariant> enrichedVariants =
+                new EnrichedStructuralVariantFactory().enrich(purpleData.allSomaticStructuralVariants());
+
+        return enrichedVariants.stream()
+                .filter(variant -> !variant.isFiltered() && variant.type() != StructuralVariantType.SGL)
+                .mapToInt(variant ->
+                {
+                    Integer count = variant.start().tumorVariantFragmentCount();
+                    return count != null ? count : 0;
+                })
+                .sum();
+    }
+
+    private static int smallVariantAlleleReadCount(@NotNull PurpleData purpleData)
+    {
+        return purpleData.allSomaticVariants().stream()
+                .filter(variant -> variant.type() == VariantType.SNP)
+                .mapToInt(variant -> variant.allelicDepth().AlleleReadCount)
+                .sum();
+    }
+
+    private static int hotspotMutationCount(@NotNull PurpleData purpleData, boolean convertGermlineToSomatic)
+    {
+        int count = (int) purpleData.allSomaticVariants().stream()
+                .filter(variant -> variant.tier() == VariantTier.HOTSPOT)
+                .count();
+
+        if(convertGermlineToSomatic && purpleData.reportableGermlineVariants() != null)
+        {
+            count += (int) purpleData.reportableGermlineVariants().stream()
+                    .filter(variant -> variant.tier() == VariantTier.HOTSPOT)
+                    .count();
+        }
```

Comment:
This sounds like a nice cleanup. However at the point of germline conversion we would be operating on the PurpleRecord instead of the loaded PurpleData, and it doesn't appear that variant.tier() is included in the model at that point. Wouldn't this require updating the orange-datamodel to include tier and if so is that worth it?

We do have access to HotSpotType though, is it valid to switch to that?

Reply:
It should be valid to switch to HotspotType, since `tier==HOTSPOT` and `hotspotType==HOTSPOT` are meant to be equivalent. 

HotspotType is determined by the `HOTSPOT` VCF tag, which is determined in Purple in `HotspotEnrichment` like this:
```
for(VariantHotspot hotspot : hotspots)
{
        if(variantStart == hotspot.position() && hotspot.ref().equals(variant.getReference().getBaseString())
        && variant.getAlternateAlleles().stream().map(Allele::getBaseString).collect(Collectors.toList()).contains(hotspot.alt()))
        {
            variant.getCommonInfo().putAttribute(HOTSPOT_FLAG, true);
            return;
        }
       ...
}
```
And the Hotspot TIER is set when the below is true in `HotspotSelector` in SAGE:
```
public boolean isHotspot(final SimpleVariant variant)
{
    final AtomicBoolean hotspotMatch = new AtomicBoolean(false);

    mHotspotSelector.select(variant.position(), variant.position(), hotspot ->
    {
        if(hotspot.alt().equals(variant.alt()) && hotspot.ref().equals(variant.ref()))
        {
            hotspotMatch.set(true);
        }
    });

    return hotspotMatch.get();
}
```
Where in this case the list of hotspots has already been filtered down to hotspots of the right chromosome.

So both just check whether there is a hotspot variant configured that matches in chromosome, position, ref and alt.